libs/gltfio/src/AssetLoader.cpp

```diff
@@ -687,6 +694,46 @@ bool FAssetLoader::createPrimitive(const cgltf_primitive* inPrim, Primitive* out
     return true;
 }
 
+void FAssetLoader::createLight(const cgltf_node* node, Entity entity) {
+    const cgltf_light* light = node->light;
+
+    // Create a name component using the light label if it exists, otherwise use the node label.
+    const char* name = light->name ? light->name : node->name;
+    if (mNameManager && name) {
+        mNameManager->addComponent(entity);
+        mNameManager->setName(mNameManager->getInstance(entity), name);
+    }
+
+    LightManager::Type type = getLightType(light->type);
+    LightManager::Builder builder(type);
+
+    builder.direction({0.0f, 0.0f, -1.0f});
```

Comment:
Is the direction controlled by a glTF transform?

Reply:
Ben if you need to apply the node transform, you can snarf some code from the top of `createEntity` (look for localTransform)

Reply:
So all this function is doing is adding a Light component to the entity. The entity should already have a transform component (which is set up at the top of `createEntity`),  the _actual_ transform of the light.

This direction of `0, 0, -1` is only needed because glTF specifies the "identity" orientation of lights as pointed down the negative z axis (as opposed to Filament's default negative y axis light orientation).

Reply:
👍

Reply:
heh, glad we kept the light direction in lights :-)